File: SwiftTest/zhiHuObtainDataOperation.m
//
//  zhiHuObtainDataOperation.m
//  SwiftTest
//
//  Created by lianpeng on 2017/2/23.
//  Copyright © 2017年 xulianpeng. All rights reserved.
//

#import "zhiHuObtainDataOperation.h"

@implementation zhiHuObtainDataOperation

- (id)initWithUrl:(NSString *)url
{
    if (self = [super init]) {
        
        self.urlStr  = url;
    }
    return self;
}
- (void)main{
    if (self.cancelled) {
        return;
    }
    
    [self obtainData];

}
- (void)obtainData{
    
    
}
@end

File: SwiftTest/ZhiHuObtainDataOperation.swift
//
//  ZhiHuObtainDataOperation.swift
//  SwiftTest
//
//  Created by lianpeng on 2017/2/23.
//  Copyright © 2017年 xulianpeng. All rights reserved.
//

import UIKit

class ZhiHuObtainDataOperation: Operation {

    var urlStr :String = ""
    
     func initWith(_ url :String){
        
     self.urlStr = url
        
        
    }
    
    override func main() {
        super.main()
        
        if (self.isCancelled) {
            return
        }
        
        //具体的操作
        doYourOperation()
        
    }
    func doYourOperation() {
        
        print("我的网址是什么啊)")
    }
}
